### PR TITLE
ChoiceBoard

### LINKED ISSUE #220: As a guardian I would like to be able to create a choice board so that I can create an activity where the citizen can make a choice between multiple activities
The guardian should be able to add multiple activities, from which the citizen can choose which one they want to do.

When the activity is chosen, it should be shown on the week plan as a regular activity/pictogram.

### PR DESCRIPTION
**Pull request by Simon97**
fixes #220 

To test this PR you must run the corresponding web_api local and use the corresponding api_client.

web_api:

 https://github.com/aau-giraf/web-api/pull/144 

api_client:

 https://github.com/aau-giraf/api_client/pull/68

### CODE REVIEW

**Review by mads-bo — COMMENTED** [review]
@aau-giraf/sw611f20 you have been assigned this. Here's your checklist.
Please use the following checklist. Feel free to check it off as you go along.

**Code Design**

- [ ] The code is in the right place? (Both in terms of folder structure and class structure)
- [ ] This code could not, to the best of my knowledge, have reused existing code
- [ ] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems
- [ ] The code does it introduce functionality that is not necessary for solving the problem

**Code Readability and Maintainability**

- [ ] Names are meaningful and self-documenting
- [ ] It is understandable, by reading the code, what it does
- [ ] The code is simple and readable. i.e. it contains no "hack", or obscure solution
- [ ] All new functionality is covered by tests
- [ ] It is understandable, by reading the tests, what they do
- [ ] The tests covers sensible cases. Both happy paths and exception paths
- [ ] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing
- [ ] Error messages in the code are understandable
- [ ] Confusing sections of code are covered by comments

**Functionality**

For this part, open the app, and test:

- [ ] The features that the code claims to implement, are actually implemented
- [ ] The code does not look like it has bugs
- [ ] All the new screens are reachable through in-app navigation

**Points for Discussion and Reflection**

- [ ] UI messages, that the user might meet, are actually helpful
- [ ] There are no errors that are bad practice. E.g. hard-coding a test database connection
- [ ] There is well written code in this. Remember to tell the developer!

**Review by mads-bo — COMMENTED** [review]
@aau-giraf/sw612f20 you have been assigned this. Here's your checklist.
Please use the following checklist. Feel free to check it off as you go along.

**Code Design**

- [ ] The code is in the right place? (Both in terms of folder structure and class structure)
- [ ] This code could not, to the best of my knowledge, have reused existing code
- [ ] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems
- [ ] The code does it introduce functionality that is not necessary for solving the problem

**Code Readability and Maintainability**

- [ ] Names are meaningful and self-documenting
- [ ] It is understandable, by reading the code, what it does
- [ ] The code is simple and readable. i.e. it contains no "hack", or obscure solution
- [ ] All new functionality is covered by tests
- [ ] It is understandable, by reading the tests, what they do
- [ ] The tests covers sensible cases. Both happy paths and exception paths
- [ ] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing
- [ ] Error messages in the code are understandable
- [ ] Confusing sections of code are covered by comments

**Functionality**

For this part, open the app, and test:

- [ ] The features that the code claims to implement, are actually implemented
- [ ] The code does not look like it has bugs
- [ ] All the new screens are reachable through in-app navigation

**Points for Discussion and Reflection**

- [ ] UI messages, that the user might meet, are actually helpful
- [ ] There are no errors that are bad practice. E.g. hard-coding a test database connection
- [ ] There is well written code in this. Remember to tell the developer!

File: lib/widgets/giraf_activity_screen_activity.dart [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  @override
+  _ActivityState createState() => _ActivityState();
+}
+
+class _ActivityState extends State<Activity> {
+  @override
+  Widget build(BuildContext context) {
+    return Container();
+  }
+}
```
Comment (antonhpalmer) [inline lib/widgets/giraf_activity_screen_activity.dart]: This seems to be unused code

File: lib/widgets/weekplan_screen_widgets/activity_card.dart [hunk 2]
```diff
@@ -97,10 +103,84 @@ class ActivityCard extends StatelessWidget {
     return true;
   }
 
+  ///This function builds the activity card
+  Widget buildChoiceboardAcivityCard(
```
Comment (antonhpalmer) [inline lib/widgets/weekplan_screen_widgets/activity_card.dart]: should be refactored
```suggestion
  Widget buildChoiceboardActivityCard(
```

**Review by Lortemanden — DISMISSED** [review]

File: lib/widgets/giraf_activity_screen_activity.dart (hunk 1 above)
Comment (Simon97) [reply lib/widgets/giraf_activity_screen_activity.dart]: is removed

File: lib/widgets/weekplan_screen_widgets/activity_card.dart (hunk 2 above)
Comment (Simon97) [reply lib/widgets/weekplan_screen_widgets/activity_card.dart]: Done

**Review by Lortemanden — APPROVED** [review]

**Comment by antonhpalmer** [comment]
There is no pictogram text under choiceboard activities resulting in the activities not having the same size on the weekplan screen. (Some standard pictogram text like ChoiceBoard should probably be added)
image

However this could probably be made into a release fix issue, because everything else seems to work fine

**Review by antonhpalmer — APPROVED** [review]

**Comment by mads-bo** [comment]
Denne PR har dependencies i api-client og web_api development branches, men bruger master. Den merges derfor selvom den fejler, det er testet på flere maskiner

**Pull request merged**